alts/build.gradle

```diff
@@ -13,35 +13,43 @@ sourceCompatibility = 1.7
 targetCompatibility = 1.7
 
 dependencies {
-    compile project(':grpc-auth'),
-            project(':grpc-core'),
+    api project(':grpc-core')
+    compileOnly libraries.javax_annotation
+    implementation project(':grpc-auth'),
             project(':grpc-grpclb'),
             project(':grpc-protobuf'),
             project(':grpc-stub'),
             libraries.lang,
-            libraries.protobuf,
             libraries.conscrypt
-    def nettyDependency = compile project(':grpc-netty')
-    compile (libraries.google_auth_oauth2_http) {
+    def nettyDependency = implementation project(':grpc-netty')
+    implementation (libraries.google_auth_oauth2_http) {
         // prefer our own versions instead of google-auth-oauth2-http's dependency
         exclude group: 'com.google.guava', module: 'guava'
         // we'll always be more up-to-date
         exclude group: 'io.grpc', module: 'grpc-context'
     }
-    compileOnly libraries.javax_annotation
-
-    shadow configurations.compile.getDependencies().minus(nettyDependency)
-    shadow project(path: ':grpc-netty-shaded', configuration: 'shadow')
-
-    testCompile project(':grpc-testing'),
+    implementation (libraries.guava) {
+        // prefer our own version from grpc-api instead of Guava's dependency
+        exclude group: 'com.google.code.findbugs', module: 'jsr305'
```

Comment:
As we talked about earlier, I really want to see excluded dependencies listed as an explicit dependency.

Reply:
Done with a util function
https://github.com/grpc/grpc-java/pull/6902/files#diff-c197962302397baf3a4cc36463dce5eaR191-R203

Exclude guava's transitive dependencies jsr305 and animal-sniffer-annotations, and always manually add them as runtimeOnly dependency. error_prone_annotations is an exception: It is also excluded but manually added not as runtimeOnly. It must alway compile with guava, otherwise users will see warning spams if guava is in the compile classpath but error_prone_annotations is not (See google/guava#2721).

Reply:
That's a nice solution.